src/sage/topology/cell_complex.py

```diff
@@ -865,7 +865,12 @@ def homology_with_basis(self, base_ring=QQ, cohomology=False):
             sage: list(H.basis(3))                                                      # needs sage.modules
             [h^{3,0}]
         """
-        from sage.homology.homology_vector_space_with_basis import HomologyVectorSpaceWithBasis
+        from sage.homology.homology_vector_space_with_basis import \
+            HomologyVectorSpaceWithBasis, HomologyVectorSpaceWithBasis_mod2
+        if base_ring == GF(2):
+            if cohomology:
+                return self.cohomology_ring(base_ring)
```

Comment:
```suggestion
        if cohomology:
            return self.cohomology_ring(base_ring)
        if base_ring == GF(2):
```
Otherwise it would not match the documentation outside of `GF(2)`.

Reply:
Right, good catch.